src/hipscat_import/pipeline.py

```diff
@@ -63,7 +63,7 @@ def pipeline_with_client(args: RuntimeArguments, client: Client):
 
 def _send_failure_email(args: RuntimeArguments, exception: Exception):
     if not args.completion_email_address:
-        raise exception
+        raise exception from None
```

Comment:
I'm a bit confused here. Why would we like to raise the same exception but unchain it? Should we raise another exception about the lack of `args.completion_email_address`?

Reply:
Fair. There's an additional method call in the stack trace because the exception isn't re-raised until the `_send_failure_email` method. I thought about adding a few more comments in the code to make that clearer, but instead re-structured the exception handling to hopefully make the intent clearer and to avoid the additional method in the stack trace.